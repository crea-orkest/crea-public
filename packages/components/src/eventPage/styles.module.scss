@use 'sass:map';
@use '../styles/tokens/breakpoints.module.scss' as breakpoints;
@use '../styles/tokens/spacing.module.scss' as spacing;
@use '../styles/tokens/layout.module.scss' as layout;
@use '../styles/tokens/colors.module.scss' as colors;
@use '../styles/tokens/radius.module.scss' as radius;

$image-width: 100%;
$image-height: calc($image-width * 1.414);
$padding-left-sidebar: map.get(spacing.$rem, 48);
$shadow: 5px 5px 20px 5px rgba(0, 0, 0, 0.1);

.container {
  display: grid;
  grid-template-areas:
    'main'
    'sidebar';
  grid-template-columns: 1fr;

  @media screen and (min-width: breakpoints.$sm) {
    grid-template-columns: 33% 1fr;
    grid-template-areas: 'sidebar main';
  }
}

.noGrid {
  grid-template-areas:
    'main'
    'sidebar';
  grid-template-columns: 1fr;
}

.header {
  grid-area: header;

  background-color: colors.$primary;
  color: colors.$secondary-light;
  padding: map.get(spacing.$rem, 64) 0;

  @media screen and (min-width: breakpoints.$sm) {
    padding: map.get(spacing.$rem, 96) 0;
  }
}

.titleContainer {
  @media screen and (min-width: breakpoints.$sm) {
    text-align: center;
  }
}

.program {
  list-style: none;
  padding: 0;
  margin: 0;
  margin-top: map.get(spacing.$rem, 16);

  @media screen and (min-width: breakpoints.$sm) {
    text-align: center;
  }
}

.programItem {
  color: colors.$background;
}

.programTitle {
  font-weight: bold;
}

.sidebar {
  grid-area: sidebar;
  padding-bottom: map.get(spacing.$rem, layout.$default-mobile-spacing);

  @media screen and (min-width: breakpoints.$sm) {
    padding-bottom: 0;
  }
}

$sidebar-translate: 50px;

.sidebarContent {
  @media screen and (min-width: breakpoints.$sm) {
    position: sticky;
    top: calc(map.get(spacing.$rem, 8) + $sidebar-translate);
    transform: translateY(calc($sidebar-translate * -1));
  }
}

.content {
  grid-area: main;

  @media screen and (min-width: breakpoints.$sm) {
    padding-left: $padding-left-sidebar;
  }
}

.imageContainer {
  border-radius: radius.$md;
  flex-shrink: 0;
  max-width: fit-content;

  @media screen and (min-width: breakpoints.$sm) {
    box-shadow: $shadow;
  }
}

.image {
  width: $image-width;
  height: $image-height;
  max-width: 100%;
  border-radius: inherit;
  object-fit: cover;
}

.persons {
  list-style: none;
  padding: 0;
  margin: 0;
  background-color: colors.$background;
  border-radius: radius.$md;

  @media screen and (min-width: breakpoints.$sm) {
    box-shadow: $shadow;
    padding: map.get(spacing.$rem, 32) map.get(spacing.$rem, 32);
  }
}

.person:not(:last-child) {
  margin-bottom: map.get(spacing.$rem, 16);

  @media screen and (min-width: breakpoints.$sm) {
    margin-bottom: map.get(spacing.$rem, 32);
  }
}

.personTitle {
  text-transform: uppercase;
}

.imageContainer + .persons {
  margin-top: map.get(spacing.$rem, 16);

  @media screen and (min-width: breakpoints.$sm) {
    margin-top: map.get(spacing.$rem, 48);
  }
}

.eventListItem {
  padding-top: map.get(spacing.$rem, layout.$default-mobile-spacing);

  @media screen and (min-width: breakpoints.$sm) {
    padding-left: map.get(spacing.$rem, 24);
    padding-top: map.get(spacing.$rem, layout.$default-desktop-spacing);
  }
}

.root .pageContent {
  padding-top: calc(
    map.get(spacing.$rem, layout.$default-mobile-spacing) - map.get(
        spacing.$rem,
        24
      )
  );

  @media screen and (min-width: breakpoints.$sm) {
    padding-top: calc(
      map.get(spacing.$rem, layout.$default-desktop-spacing) - map.get(
          spacing.$rem,
          32
        )
    );
  }
}

.locations {
  grid-area: footer;
  display: flex;
  flex-direction: column;
  gap: map.get(spacing.$rem, 32);

  @media screen and (min-width: breakpoints.$sm) {
    flex-direction: row;
    gap: 0;
  }
}
