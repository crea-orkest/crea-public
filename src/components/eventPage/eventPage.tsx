import Image from 'next/image'
import classNames from 'classnames'
import type { Event } from 'graphql/types/event'
import { PageContent } from 'components/pageContent'
import { LocationDetail } from 'components/locationDetail'
import { EventListItem } from 'components/eventListItem'

import styles from './styles.module.scss'

export interface Props {
  data: Event
}

export const EventPage = ({ data }: Props) => {
  return (
    <article className={styles.root}>
      <header
        className={classNames(styles.header, 'with-background', {
          [`${styles.withImage}`]: data.image?.url,
        })}
      >
        {data.image?.url && (
          <div className={classNames(styles.headerImageWrapper)}>
            <Image
              className={classNames(styles.headerImage)}
              src={data.image.url}
              alt={data.image.alt}
              width={data.image.width ?? 100}
              height={data.image.height ?? 100}
            />
            <div className={classNames(styles.background)} />
          </div>
        )}

        <div className={classNames(styles.titleContainer, 'content-layout')}>
          <h2>{data.title}</h2>

          {data.program.length > 0 && (
            <ul className={styles.program}>
              {data.program.map((item) => {
                if (!item?.id) return null
                return (
                  <li
                    key={item.id}
                    className={classNames(styles.programItem, 'text-large')}
                  >
                    <span className={styles.programTitle}>{item.composer}</span>{' '}
                    â€” {item.title}
                  </li>
                )
              })}
            </ul>
          )}
        </div>
      </header>
      <div
        className={classNames(styles.container, 'content-layout', {
          [`${styles.noGrid}`]: !data.image?.url,
        })}
      >
        <aside className={styles.sidebar}>
          <div className={styles.sidebarContent}>
            {data.image?.url && (
              <div className={classNames(styles.imageContainer)}>
                <Image
                  priority={true}
                  className={classNames(styles.image)}
                  src={data.image.url}
                  alt={data.image.alt}
                  width={data.image.width ?? 100}
                  height={data.image.height ?? 100}
                />
              </div>
            )}

            {data.persons.length > 0 && (
              <ul className={styles.persons}>
                {data.persons.map((person) => {
                  if (!person?.id) return null
                  return (
                    <li key={person.id} className={styles.person}>
                      <p
                        className={classNames(styles.personTitle, 'text-small')}
                      >
                        {person.description}
                      </p>
                      <p className="h5">{person.name}</p>
                    </li>
                  )
                })}
              </ul>
            )}
          </div>
        </aside>
        <div className={styles.content}>
          {data.locations.length > 0 && (
            <div className={styles.eventListItem}>
              <EventListItem
                data={data}
                size="large"
                isLast={true}
                title="Agenda"
                showLink={false}
                showImage={false}
                showProgram={false}
              />
            </div>
          )}

          <PageContent
            sectionClassName={styles.pageContent}
            items={data.content}
          />
        </div>
      </div>
      <div className={styles.locations}>
        {data.locations.map((location) => {
          if (!location?.id || !location.startTime) return null
          return <LocationDetail key={location?.id} data={location} />
        })}
      </div>
    </article>
  )
}
