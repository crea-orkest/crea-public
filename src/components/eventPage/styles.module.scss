@use 'sass:map';
@use '../../styles/tokens/breakpoints.scss' as breakpoints;
@use '../../styles/tokens/spacing.scss' as spacing;
@use '../../styles/tokens/layout.scss' as layout;
@use '../../styles/tokens/colors.scss' as colors;
@use '../../styles/tokens/radius.scss' as radius;

$image-width: 100%;
$image-height: calc($image-width * 1.414);
$padding-left-sidebar: map.get(spacing.$rem, 48);

.container {
  display: grid;
  grid-template-areas:
    'main'
    'sidebar';
  grid-template-columns: 1fr;

  @media screen and (min-width: breakpoints.$sm) {
    grid-template-columns: 33% 1fr;
    grid-template-areas: 'sidebar main';
  }
}

.noGrid {
  grid-template-areas:
    'main'
    'sidebar';
  grid-template-columns: 1fr;
}

.header {
  grid-area: header;

  background-color: colors.$primary;
  color: colors.$secondary-light;
  padding: map.get(spacing.$rem, 64) 0;

  @media screen and (min-width: breakpoints.$sm) {
    padding: map.get(spacing.$rem, 96) 0;
  }
}

.titleContainer {
  @media screen and (min-width: breakpoints.$sm) {
    text-align: center;
  }
}

.sidebar {
  grid-area: sidebar;
  padding-bottom: map.get(spacing.$rem, layout.$default-mobile-spacing);

  @media screen and (min-width: breakpoints.$sm) {
    padding-bottom: 0;
  }
}

$sidebar-translate: 50px;

.sidebarContent {
  @media screen and (min-width: breakpoints.$sm) {
    position: sticky;
    top: calc(map.get(spacing.$rem, 8) + $sidebar-translate);
    transform: translateY(calc($sidebar-translate * -1));
  }
}

.content {
  grid-area: main;

  @media screen and (min-width: breakpoints.$sm) {
    padding-left: $padding-left-sidebar;
  }
}

.imageContainer {
  border-radius: radius.$md;
  flex-shrink: 0;
  max-width: fit-content;

  @media screen and (min-width: breakpoints.$sm) {
    box-shadow: 6px 6px 15px 0px rgba(0, 0, 0, 0.25);
  }
}

.image {
  width: $image-width;
  height: $image-height;
  max-width: 100%;
  border-radius: inherit;
  object-fit: cover;
}

.eventListItem {
  padding-top: map.get(spacing.$rem, layout.$default-mobile-spacing);

  @media screen and (min-width: breakpoints.$sm) {
    padding-left: map.get(spacing.$rem, 24);
    padding-top: map.get(spacing.$rem, layout.$default-desktop-spacing);
  }
}

.root .pageContent {
  padding-top: calc(
    map.get(spacing.$rem, layout.$default-mobile-spacing) -
      map.get(spacing.$rem, 24)
  );

  @media screen and (min-width: breakpoints.$sm) {
    padding-top: calc(
      map.get(spacing.$rem, layout.$default-desktop-spacing) -
        map.get(spacing.$rem, 32)
    );
  }
}

.locations {
  grid-area: footer;
  display: flex;
  flex-direction: column;
  gap: map.get(spacing.$rem, 32);

  @media screen and (min-width: breakpoints.$sm) {
    flex-direction: row;
    gap: 0;
  }
}
