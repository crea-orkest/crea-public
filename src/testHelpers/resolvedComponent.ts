export const resolvedComponent = async <T>(
  Component: Function,
  props: T
): Promise<() => JSX.Element> => {
  const ComponentResolved = await Component(props);
  return () => ComponentResolved;
};
