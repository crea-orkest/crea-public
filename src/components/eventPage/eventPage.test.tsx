import { EventPage } from './eventPage'
import type { Props } from './eventPage'
import React from 'react'
import { resolvedComponent } from '../../utils/testHelpers/resolvedComponent'
import { render } from '@testing-library/react'

jest.mock('../../graphql/getters/getEventPage', () => {
  const originalModule = jest.requireActual(
    '../../graphql/getters/getEventPage'
  )
  return {
    __esModule: true,
    ...originalModule,
    getEventPage: jest.fn(),
  }
})

describe('Event page', () => {
  it('shows all the data', async () => {
    const Resolved = await resolvedComponent<Props>(EventPage, {
      data: {
        id: 'unique-id',
        title: 'title',
        image: null,
        locations: [],
        persons: [],
        program: [],
        url: '/some-slug',
        content: [],
      },
    })

    const { container } = render(<Resolved />)

    expect(container).toMatchSnapshot()
  })
})
