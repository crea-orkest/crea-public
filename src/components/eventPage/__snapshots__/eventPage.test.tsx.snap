// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Event page shows all the data 1`] = `
<div>
  <article
    class="root"
  >
    <header
      class="header with-background"
    >
      <div
        class="titleContainer content-layout"
      >
        <h2>
          title
        </h2>
      </div>
    </header>
    <div
      class="container content-layout noGrid"
    >
      <aside
        class="sidebar"
      >
        <div
          class="sidebarContent"
        />
      </aside>
      <div
        class="content"
      />
    </div>
    <div
      class="locations"
    />
  </article>
</div>
`;
